build/flaws.js

```diff
@@ -22,26 +26,134 @@ const {
 const { humanFileSize } = require("./utils");
 const { VALID_MIME_TYPES } = require("../filecheck/constants");
 
-function injectFlaws(doc, $, options, { rawContent }) {
+function injectFlaws(doc, $, options, document) {
   if (doc.isArchive) return;
 
-  injectBrokenLinksFlaws(
-    options.flawLevels.get("broken_links"),
-    doc,
-    $,
-    rawContent
-  );
+  const flawChecks = [
+    ["unsafe_html", injectUnsafeHTMLFlaws, false],
+    ["broken_links", injectBrokenLinksFlaws, true],
+    ["bad_bcd_queries", injectBadBCDQueriesFlaws, false],
+    ["bad_pre_tags", injectPreTagFlaws, false],
+    ["heading_links", injectHeadingLinksFlaws, false],
+  ];
+
```

Comment:
The reason this PR is so much bigger and touches so many things is because I'm sneaking in a refactoring. 
It used to be like this:

```javascript
injectSomeFlaw(level, doc, $, options)

...

function injectSomeFlaw(level, doc, $, options) {
  if (level === IGNORE) return;
  
  ...

  if (level === ERROR) throw new Error(...)
}
```

Now, I'm collecting a list of all `inject*` functions and taking care of the level checking BEFORE even call it. Now, each individual `inject*` function can be simplified.